geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/ListRegionCommand.java

```
@@ -50,61 +48,52 @@
 
-        Set<String> regionNames = new TreeSet<>();
+    if (resultList != null) {
+      // Gather all RegionInformation into a flat set.
+      regionInfoSet.addAll(resultList.stream().filter(Objects::nonNull)
+          .filter(Object[].class::isInstance).map(Object[].class::cast).flatMap(Arrays::stream)
+          .filter(RegionInformation.class::isInstance).map(RegionInformation.class::cast)
+          .collect(Collectors.toSet()));
 
-        for (RegionInformation regionInfo : regionInfoSet) {
-          regionNames.add(regionInfo.getName());
-          Set<String> subRegionNames = regionInfo.getSubRegionNames();
+      Set<String> regionNames = new TreeSet<>();
 
-          regionNames.addAll(subRegionNames);
-        }
+      for (RegionInformation regionInfo : regionInfoSet) {
+        regionNames.add(regionInfo.getName());
+        Set<String> subRegionNames = regionInfo.getSubRegionNames();
 
-        for (String regionName : regionNames) {
-          resultData.accumulate("List of regions", regionName);
-        }
+        regionNames.addAll(subRegionNames);
+      }
 
-        if (!regionNames.isEmpty()) {
-          result = ResultBuilder.buildResult(resultData);
+      if (regionNames.isEmpty()) {
+        return ResultModel.createInfo(CliStrings.LIST_REGION__MSG__NOT_FOUND);
+      }
 
-        } else {
-          result = ResultBuilder.createInfoResult(CliStrings.LIST_REGION__MSG__NOT_FOUND);
-        }
+      for (String regionName : regionNames) {
+        resultData.accumulate("List of regions", regionName);
       }
-    } catch (FunctionInvocationTargetException e) {
-      result = ResultBuilder.createGemFireErrorResult(CliStrings
-          .format(CliStrings.COULD_NOT_EXECUTE_COMMAND_TRY_AGAIN, CliStrings.LIST_REGION));
-    } catch (Exception e) {
-      result = ResultBuilder
-          .createGemFireErrorResult(CliStrings.LIST_REGION__MSG__ERROR + " : " + e.getMessage());
     }
```

why do we no longer need exception handling here?

---

the CommandExecutor will catch all exceptions and do these things.

---

cool!